packages/dds/matrix/src/test/memory/matrix.spec.ts

```diff
@@ -1,155 +0,0 @@
-/*!
- * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
- * Licensed under the MIT License.
- */
-
-import {
-	type IMemoryTestObject,
-	benchmarkMemory,
-	isInPerformanceTestingMode,
-} from "@fluid-tools/benchmark";
-import type { IChannel } from "@fluidframework/datastore-definitions/internal";
-import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
-
-import { SharedMatrix, type ISharedMatrix } from "../../index.js";
-import { matrixFactory } from "../utils.js";
-
-function createLocalMatrix(id: string): ISharedMatrix & IChannel {
-	return matrixFactory.create(new MockFluidDataStoreRuntime(), id);
-}
-
-describe("Matrix memory usage", () => {
-	// IMPORTANT: variables scoped to the test suite are a big problem for memory-profiling tests
-	// because they won't be out of scope when we garbage-collect between runs of the same test,
-	// and that will skew measurements. Tests should allocate all the memory they need using local
-	// variables scoped to the test function itself, so several iterations of a given test can
-	// measure from the same baseline (as much as possible).
-
-	beforeEach(async () => {
-		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
-		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
-		// See the comment at the top of the test suite for more details.
-	});
-
-	afterEach(() => {
-		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
-		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
-		// See the comment at the top of the test suite for more details.
-	});
-
-	describe("Detached Matrix", () => {
-		benchmarkMemory(
-			new (class implements IMemoryTestObject {
-				readonly title = "Create empty Matrix";
-				minSampleCount = 500;
-
-				private localMatrix: SharedMatrix = createLocalMatrix("testLocalMatrix");
-
-				async run(): Promise<void> {
-					this.localMatrix = createLocalMatrix("testLocalMatrix");
-				}
-			})(),
-		);
-
-		const numbersOfEntriesForTests = isInPerformanceTestingMode
-			? [100, 1000, 5000]
-			: // When not measuring perf, use a single smaller data size so the tests run faster.
-				[10];
-
-		for (const x of numbersOfEntriesForTests) {
-			benchmarkMemory(
-				new (class implements IMemoryTestObject {
-					readonly title = `Insert and remove a column ${x} times`;
```

Comment:
I think all the existing tests for `insert + remove <something>` check for something that the new ones miss, namely "is the memory usage close to 0 when we add things and immediately remove them". Did we deliberately decide that kind of check isn't necessary?

Reply:
Good point. I would recommend we do that as a more targeted test, though (for clarity). If that scenario was the purpose of this test, that was not clear. I would prefer the replacement test be explicitly expressed in terms of measuring the memory delta after removing freshly inserted rows/columns. The test can simply insert x rows then delete x rows (rather than inserting and deleting 1 at a time).

Reply:
Note from Sync-up with Josh: We decided to remove the existing test and create a new one. The main reason is that the current test mixes insert and remove operations within a single case, making it difficult to isolate and measure the memory usage of each operation individually. Additionally, the test does not allow the matrix size to grow significantly, which means it fails to cover some important test criteria.